### ISSUE TITLE
Add assertThat for java.time.Period

### ISSUE THREAD

**Issue by eddumelendez** [joel-costigliola/assertj-core#1853]
#### Summary

Similar to #1650 it would be great to support `java.time.Period`

#### Example

```java
Period period = Period.of(1,1,1)
assertThat(period)
    .hasYear(1)
    .hasMonth(1)
    .hasDays(1);
```

If you are agree with this I can work on it.

**Comment by joel-costigliola** [comment]
Goo idea, go for it @eddumelendez :)

**Comment by eddumelendez** [comment]
Resolved by #1902
